systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java

```diff
@@ -0,0 +1,822 @@
+        assertThat(received, is(20));
+    }
+
+    @Test
+    @Tag(NODEPORT_SUPPORTED)
+    void testCustomChainCertificatesForNodePort() throws Exception {
+        String topicName = "test-topic-" + rng.nextInt(Integer.MAX_VALUE);
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 3)
+            .editSpec()
+                .editKafka()
+                    .editListeners()
+                        .withNewTls()
+                            .withNewConfiguration()
+                                .withNewBrokerCertChainAndKey()
+                                    .withSecretName(customCertChain1)
+                                    .withKey("ca.key")
+                                    .withCertificate("ca.crt")
+                                .endBrokerCertChainAndKey()
+                            .endConfiguration()
+                        .endTls()
+                        .withNewKafkaListenerExternalNodePort()
+                            .withNewConfiguration()
+                                .withNewBrokerCertChainAndKey()
+                                    .withSecretName(customCertChain1)
+                                    .withKey("ca.key")
+                                    .withCertificate("ca.crt")
+                                .endBrokerCertChainAndKey()
+                            .endConfiguration()
+                        .endKafkaListenerExternalNodePort()
+                    .endListeners()
+                .endKafka()
+            .endSpec().done();
+
+        String userName = "alice";
+        KafkaUser aliceUser = KafkaUserResource.tlsUser(CLUSTER_NAME, userName).done();
+
+        kafkaClient.setCaCertName(customRootCA1);
+        Future producer = kafkaClient.sendMessagesTls(topicName, NAMESPACE, CLUSTER_NAME, userName, 10, "SSL");
+        Future consumer = kafkaClient.receiveMessagesTls(topicName, NAMESPACE, CLUSTER_NAME, userName, 10, "SSL");
+
+        assertThat("Producer didn't produce all messages", producer.get(1, TimeUnit.MINUTES), is(10));
+        assertThat("Consumer didn't consume all messages", consumer.get(1, TimeUnit.MINUTES), is(10));
+
+        // Deploy client pod with custom certificates and collect messages from internal TLS listener
+        KafkaClientsResource.deployKafkaClients(true, CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS, CLUSTER_NAME, NAMESPACE, false, aliceUser).done();
+
+        externalKafkaClient.setPodName(kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName());
+        int sent = externalKafkaClient.sendMessagesTls(topicName, NAMESPACE, CLUSTER_NAME, userName, 10, "TLS");
+        assertThat(sent, is(10));
+        int received = externalKafkaClient.receiveMessagesTls(topicName, NAMESPACE, CLUSTER_NAME, userName, 20, "TLS", "consumer-group-certs-2");
+        assertThat(received, is(20));
```

Comment:
Would it be good to create method `verifyCommunicationWIthExternalClient` and change it in all test cases? Same method applicable in internal.